app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -33,27 +39,38 @@ class WireActivityViewModel @Inject constructor(
 
     private val isUserLoggedIn = currentSession != null
     var serverConfig: ServerConfig = ServerConfig.DEFAULT
+    private var ssoDeepLinkResult: DeepLinkResult.SSOLogin? = null
+
+    fun navigationArguments() =
+        if (ssoDeepLinkResult != null) {
+            listOf(serverConfig, ssoDeepLinkResult!!)
+        } else listOf(serverConfig)
 
     fun startNavigationRoute() = when {
+        ssoDeepLinkResult is DeepLinkResult.SSOLogin -> NavigationItem.Login.getRouteWithArgs()
         serverConfig.apiBaseUrl != ServerConfig.DEFAULT.apiBaseUrl -> NavigationItem.Login.getRouteWithArgs()
         isUserLoggedIn -> NavigationItem.Home.getRouteWithArgs()
         else -> NavigationItem.Welcome.getRouteWithArgs()
     }
 
     private fun loadServerConfig(url: String) = runBlocking {
-        return@runBlocking when (val result = getServerConfigUserCase.invoke(url)) {
+        return@runBlocking when (val result = getServerConfigUseCase.invoke(url)) {
```

Comment:
The invoke call can be deleted
```suggestion
        return@runBlocking when (val result = getServerConfigUseCase(url)) {
```